### PR TITLE
Removes router information from messages sent via websocket

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Removes usage of router server host and port in messages sent via websocket
- Fixes node proxy to accept different `requestOrigin` parameters to query router vs market origins. This is used to query market info as it comes from a different origin.
- Fixes `/forwardMarketToCdap` and `/downloadLogs` api endpoints to accept path instead of entire url

Based on the changes a request looks like,

```
{
  "id": <uuid>,
  "json": true,
  "method": "GET",
  "suppressErrors": false,
  "url": "/v3/namespaces/default/apps/SamplePipeline/workflows/DataPipelineWorkflow/runs"
}
```

Build: https://builds.cask.co/browse/CDAP-UDUT196

### CODE REVIEW

File: cdap-ui/app/cdap/components/HttpExecutor/InputPath/index.js [hunk 1]
```diff
@@ -43,21 +43,13 @@ const mapDispatch = (dispatch) => {
...
-    '/v3/',
-  ].join('');
-
   return (
     <div className="input-path-container">
       <div className="input-group">
         <div className="input-group-prepend">
-          <div className="input-group-text">{url}</div>
+          <div className="input-group-text">
+            {`\${router-procol}://\${router-host}:\${router-port}/v3/`}
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/HttpExecutor/InputPath/index.js]: typo: `router-protocol`

File: cdap-ui/app/services/data/cdap-url.js [hunk 2]
```diff
@@ -39,15 +39,7 @@ angular.module(PKG.name + '.services')
...
-        url = [
-          MY_CONFIG.sslEnabled? 'https://': 'http://',
-          MY_CONFIG.cdap.routerServerUrl,
-          ':',
-          MY_CONFIG.sslEnabled? MY_CONFIG.cdap.routerSSLServerPort: MY_CONFIG.cdap.routerServerPort,
-          resource._cdapPathV2 ? '/v2' : '/v3',
-          resource._cdapPath || resource._cdapPathV2
-        ].join('');
-
+        url = resource._cdapPath || resource._cdapPathV2;
```
Comment (elfenheart) [inline cdap-ui/app/services/data/cdap-url.js]: we don't have anymore v2, do we?

Comment (ajainarayanan) [reply cdap-ui/app/services/data/cdap-url.js]: Don't think so. Have removed it.

File: cdap-ui/app/cdap/api/market.js [hunk 3]
```diff
@@ -18,16 +18,22 @@ import DataSourceConfigurer from 'services/datasource/DataSourceConfigurer';
 import { apiCreatorAbsPath } from 'services/resource-helper';
 
 let dataSrc = DataSourceConfigurer.getInstance();
-const basepath = `${window.CDAP_CONFIG.marketUrl}`;
+const basepath = window.CDAP_CONFIG.marketUrl;
+// FIXME: Right now this is scattered across node and client. Need to consolidate this.
```
Comment (chtyim) [inline cdap-ui/app/cdap/api/market.js]: Open a JIRA, label it with techdebt and have the JIRA number in the comment.

File: cdap-ui/server/express.js [hunk 4]
```diff
@@ -193,18 +194,13 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
...
-        routerServerPort: cdapConfig['router.server.port'],
-        routerSSLServerPort: cdapConfig['router.ssl.server.port'],
         standaloneWebsiteSDKDownload:
           uiSettings['standalone.website.sdk.download'] === 'true' || false,
         uiDebugEnabled: uiSettings['ui.debug.enabled'] === 'true' || false,
       },
       hydrator: {
         previewEnabled: cdapConfig['enable.preview'] === 'true',
       },
-      marketUrl: cdapConfig['market.base.url'],
```
Comment (elfenheart) [inline cdap-ui/server/express.js]: I don't think you can remove this. The market images are using this path I think?

Comment (ajainarayanan) [reply cdap-ui/server/express.js]: Good point. Reverting.

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by ajainarayanan** [comment]
https://builds.cask.co/browse/CDAP-UDUT196-7 Latest green build.

**Pull request merged**